src/Sitecore.AspNetCore.SDK.Pages/Controllers/PagesSetupController.cs

```diff
@@ -0,0 +1,149 @@
+                SetConfigResponseHeaders(Response);
+                return Ok(BuildConfigResponseBody());
+            }
+
+            return BadRequest();
+        }
+
+        /// <summary>
+        /// The render endpoint used to redirect the Pages application to the correct page.
+        /// </summary>
+        /// <returns>Redirect response.</returns>
+        [HttpGet]
+        public IActionResult Render()
+        {
+            if (IsValidPagesRenderRequest(Request))
+            {
+                logger.LogDebug(Resources.Debug_ProcessingValidPagesRenderRequest);
+                PagesRenderArgs args = ParseQueryStringArgs(Request);
+                return Redirect($"{args.Route}?mode={args.Mode}&sc_itemid={args.ItemId}&sc_version={args.Version}&sc_lang={args.Language}&sc_site={args.Site}&sc_layoutKind={args.LayoutKind}&secret={args.EditingSecret}&tenant_id={args.TenantId}&route={args.Route}");
+            }
+
+            return BadRequest();
+        }
+
+        private PagesRenderArgs ParseQueryStringArgs(HttpRequest request)
+        {
+            return new PagesRenderArgs
+            {
+                ItemId = Guid.TryParse(request.Query["sc_itemid"].FirstOrDefault(), out Guid itemId) ? itemId : Guid.Empty,
+                EditingSecret = request.Query["secret"].FirstOrDefault() ?? string.Empty,
+                Language = request.Query["sc_lang"].FirstOrDefault() ?? string.Empty,
+                LayoutKind = request.Query["sc_layoutKind"].FirstOrDefault() ?? string.Empty,
+                Mode = request.Query["mode"].FirstOrDefault() ?? string.Empty,
+                Route = request.Query["route"].FirstOrDefault() ?? string.Empty,
+                Site = request.Query["sc_site"].FirstOrDefault() ?? string.Empty,
+                Version = int.TryParse(request.Query["sc_version"].FirstOrDefault(), out int version) ? version : 0,
+                TenantId = request.Query["tenant_id"].FirstOrDefault() ?? string.Empty,
+            };
+        }
+
+        private bool IsValidPagesRenderRequest(HttpRequest httpRequest)
+        {
+            ArgumentNullException.ThrowIfNull(httpRequest);
+
+            if (!IsValidEditingSecret(httpRequest))
+            {
+                logger.LogError(Resources.Error_InvalidPagesEditingSecretValue);
+                return false;
+            }
+
+            return true;
+        }
+
+        private PagesConfigResponse BuildConfigResponseBody()
+        {
+            return new PagesConfigResponse
+            {
+                EditMode = "metadata",
+                Components = renderingEngineOptions.RendererRegistry.Where(x => x.Value.ComponentName != string.Empty).Select(x => x.Value.ComponentName).ToList()
+            };
+        }
+
+        private void SetConfigResponseHeaders(HttpResponse httpResponse)
+        {
+            httpResponse.Headers.ContentSecurityPolicy = $"frame-ancestors 'self' {options.ValidOrigins} {options.ValidEditingOrigin}";
```

Comment:
How will this work together with multisite? It looks like all origins needs to be specified in configuration.
When using ``UseMultisite`` I would expect that it should use the host of the the resolved site

Reply:
This param currently isn't being populated by JSS, they return an empty string which is what we're doing as well here as the options param doesn't have a default value for this prop.

The idea is that the `PagesOptions` class will work without OOTB without the need to override any settings, but giving the ability to do that down the track if something on the pages side changes and we need to.